src/kernels/ipywidgets/commonMessageCoordinator.ts

```diff
@@ -148,6 +192,14 @@ export class CommonMessageCoordinator {
         this.getIPyWidgetScriptSource().onMessage(message, payload);
     }
 
+    private sendPendingWebViewMessages() {
+        if (!this.webview || !this.webview.isReady || !this.readyMessageReceived) {
+            return;
+        }
+        while (this.queuedMessages.length) {
```

Comment:
`while` will lock the thread. Could there be many queued messages? It might be better to use `queuedMessages.forEach`

Reply:
both will lock the thread.
only async code did not block threads in v8.
l running such code in a tight loop is ok, we won't have millions, it's a very very small data set and execution, which will not be cpu intensive